ui/timeline/src/commonMain/kotlin/social/androiddev/timeline/navigation/TimelineComponent.kt

```diff
@@ -9,7 +9,67 @@
  */
 package social.androiddev.timeline.navigation
 
+import com.arkivanov.decompose.ComponentContext
+import com.arkivanov.essenty.instancekeeper.InstanceKeeper
+import com.arkivanov.essenty.instancekeeper.getOrCreate
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.collectIndexed
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.mobilenativefoundation.store.store5.MarketResponse
+import social.androiddev.common.repository.timeline.HomeTimelineRepository
+import social.androiddev.common.timeline.TimelineItem
+import kotlin.coroutines.CoroutineContext
+
 /**
  * The base component describing all business logic needed for the timeline view
  */
-interface TimelineComponent
+interface TimelineComponent{
+    val state: StateFlow<MarketResponse<List<TimelineItem>>>
+}
+
+class DefaultTimelineComponent(
```

Comment:
I initially had these in same file, and got feedback to move the default impls to their own file (I forget the additional reasoning). But, let's be consistent with whatever we decide